File: blog-web-app/typescript/tests/FixedClock.ts
import type { Clock } from '../src/Clock.js';

export class FixedClock implements Clock {
  private _current: Date;

  constructor(current: Date) {
    this._current = current;
  }

  now(): Date {
    return this._current;
  }

  advanceTo(date: Date): void {
    this._current = date;
  }

  advanceByMinutes(minutes: number): void {
    this._current = new Date(this._current.getTime() + minutes * 60_000);
  }
}

File: blog-web-app/python/tests/fixed_clock.py
from __future__ import annotations

from datetime import datetime, timedelta


class FixedClock:
    def __init__(self, current: datetime) -> None:
        self._current = current

    def now(self) -> datetime:
        return self._current

    def advance_to(self, dt: datetime) -> None:
        self._current = dt

    def advance_by_minutes(self, minutes: int) -> None:
        self._current = self._current + timedelta(minutes=minutes)
